websub-generic-integration-tests/tests/test_content_negotiation.bal

```diff
@@ -0,0 +1,167 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/websub;
+import ballerina/test;
+
+websub:Hub WebSubHub = startHubAndRegisterTopic();
+listener http:Listener httpListener = new (24080);
+
+service /publisherService on httpListener {
+    resource function get discoveryWithAcceptAndAcceptLanguage(http:Caller caller, http:Request req) {
+        http:Response response = new;
+        string media_type = req.getHeader("Accept");
+        string language_type = req.getHeader("Accept-Language");
+        if (media_type == "application/json" && language_type == "de-DE") {
+            websub:addWebSubLinkHeader(response, [WebSubHub.subscriptionUrl], WEBSUB_TOPIC_ONE);
+            response.statusCode = 202;
+            var result = caller->respond(response);
+            if (result is error) {
+                log:printError("Error responding ", err = result);
+            }
+        } else {
+            response.statusCode = 406;
+            var res = caller->respond(response);
+        }
+    }
+
+    //Publisher need accept header only and agree with application/json as Accept header
+    resource function get discoveryAcceptHeaderOnly(http:Caller caller, http:Request req) {
+        http:Response response = new;
+        string media_type = req.getHeader("Accept");
+        if (media_type == "application/json") {
+            websub:addWebSubLinkHeader(response, [WebSubHub.subscriptionUrl], WEBSUB_TOPIC_ONE);
+            response.statusCode = 202;
+            var result = caller->respond(response);
+            if (result is error) {
+                log:printError("Error responding ", err = result);
+            }
+        } else {
+            response.statusCode = 406;
+            var res = caller->respond(response);
+        }
+    }
+
+    //Publisher need accept language header only and agree with de-DE as Accept-Language header
+    resource function get discoveryAcceptLanguageOnly(http:Caller caller, http:Request req) {
+        http:Response response = new;
+        string language_type = req.getHeader("Accept-Language");
+        if (language_type == "de-DE") {
+            websub:addWebSubLinkHeader(response, [WebSubHub.subscriptionUrl], WEBSUB_TOPIC_ONE);
+            response.statusCode = 202;
+            var result = caller->respond(response);
+            if (result is error) {
+                log:printError("Error responding ", err = result);
+            }
+        } else {
+            response.statusCode = 406;
+            var res = caller->respond(response);
+        }
+    }
+}
+
+@test:Config{}
+//SubscriberServiceConfig's accept and acceptLanguage field values match with publisher's acceptable Accept and Accept-Language header values.
```

Comment:
Shall we move these comments above the `@test:Config` annotation in each test?

Leave a space after `//`.

Please check for all tests here.